### PR TITLE
feat: update qdrant version to 0.10.1 

### PR DESCRIPTION
**Pull request by AnneYang720**
This PR is based on the work of #589 

Goals:

- [x] update qdrant version to 0.10.1
- [x] update some of the tests (add grpc tests for qdrant)
- [x] update docs

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -238,7 +238,7 @@ jobs:
           pytest --suppress-no-test-exit-code --cov=docarray --cov-report=xml \
             -v -s -m "not gpu" ${{ matrix.test-path }}
           echo "::set-output name=codecov_flag::docarray"
-        timeout-minutes: 30
+        timeout-minutes: 40
```
Comment (JoanFM) [inline .github/workflows/ci.yml:241]: why is this needed now?

Comment (AnneYang720) [reply .github/workflows/ci.yml:241]: Currently `tests/unit/array/mixins/oldproto` takes more than 30 mins to finish

File: docarray/array/storage/qdrant/backend.py [hunk 2]
```diff
@@ -47,6 +53,11 @@ class QdrantConfig:
 
 
 class BackendMixin(BaseBackendMixin):
+    @property
+    @abstractmethod
+    def client(self) -> 'QdrantClient':
```
[the remark this answers is not in the document]
Comment (AnneYang720) [reply docarray/array/storage/qdrant/backend.py:58]: This is implemented in `docarray/array/storage/qdrant/__init__.py`.

File: scripts/docker-compose.yml [hunk 3]
```diff
@@ -10,9 +10,10 @@ services:
       AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
       PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
   qdrant:
-    image: qdrant/qdrant:v0.8.0
+    image: qdrant/qdrant:v0.10.1
     ports:
-      - "41233:41233"
+      - "41237:41237"
+      - "41238:41238"
```
[the remark this answers is not in the document]
Comment (AnneYang720) [reply scripts/docker-compose.yml]: fixed. keep both ports for grpc but fix the ports inside docker

File: docs/advanced/document-store/qdrant.md [hunk 4]
```diff
@@ -205,57 +213,4 @@ Embeddings Nearest Neighbours with "price" at most 7:
 	embedding=[6. 6. 6.],	 price=6
 	embedding=[5. 5. 5.],	 price=5
 	embedding=[4. 4. 4.],	 price=4
-```
-### Example of `.filter` with a filter
```
Comment (JoanFM) [inline docs/advanced/document-store/qdrant.md:209]: Why is this removed?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/qdrant.md [hunk 5]
```diff
@@ -79,13 +80,19 @@ The following configs can be set:
...
-| `scroll_batch_size`   | batch size used when scrolling over the storage                                                                                              | 64                                               |
-| `ef_construct`        | Number of neighbours to consider during the index building.  Larger the value - more accurate the search, more time required to build index. | `None`, defaults to the default value in Qdrant* |
-| `full_scan_threshold` | Minimal amount of points for additional payload-based indexing.                                                                              | `None`, defaults to the default value in Qdrant*                                               |
-| `m`                   | Number of edges per node in the index graph. Larger the value - more accurate the search, more space required.                               | `None`, defaults to the default value in Qdrant*                                               |
+| `distance`            | Distance metric to be used during search. Can be 'cosine', 'dot' or 'euclidean'                                                              | `'cosine'`                                       |
```
Comment (alexcg1) [inline docs/advanced/document-store/qdrant.md:83]:
```suggestion
| `distance`            | Distance metric to use during search. Can be 'cosine', 'dot' or 'euclidean'                                                              | `'cosine'`                                       |
```

File: docs/advanced/document-store/qdrant.md [hunk 6]
```diff
@@ -79,13 +80,19 @@ The following configs can be set:
...
+| `host`                | Hostname of the Qdrant server                                                                                                                | `'localhost'`                                    |
+| `port`                | Port of the Qdrant server                                                                                                                    | `6333`                                           |
+| `grpc_port`           | Port of the Qdrant gRPC interface                                                                                                            | `6334`                                           |
+| `prefer_grpc`         | Set `true` to use gPRC interface whenever possible in custom methods                                                                         | `False`                                          |
```
Comment (alexcg1) [inline docs/advanced/document-store/qdrant.md:87]:
```suggestion
| `prefer_grpc`         | Set `True` to use gPRC interface whenever possible in custom methods                                                                         | `False`                                          |
```

Comment (alexcg1) [reply docs/advanced/document-store/qdrant.md:87]: It should be capitalized, right? If so, fix this throughout the doc

File: docs/advanced/document-store/qdrant.md [hunk 7]
```diff
@@ -79,13 +80,19 @@ The following configs can be set:
...
+| `api_key`             | API key for authentication in Qdrant Cloud                                                                                                   | `None`                                           |
+| `https`               | Set `true` to use HTTPS(SSL) protocol                                                                                                        | `None`                                           |
+| `serialize_config`    | [Serialization config of each Document](../../../fundamentals/document/serialization.md)                                                     | `None`                                           |
+| `scroll_batch_size`   | Batch size used when scrolling over the storage                                                                                              | `64`                                             |
+| `ef_construct`        | Number of neighbours to consider during the index building.  Larger the value - more accurate the search, more time required to build index  | `None`, defaults to the default value in Qdrant* |
```
Comment (alexcg1) [inline docs/advanced/document-store/qdrant.md:92]:
```suggestion
| `ef_construct`        | Number of neighbours to consider during the index building.  Larger = more accurate search, more time to build index  | `None`, defaults to the default value in Qdrant* |
```

File: docs/advanced/document-store/qdrant.md [hunk 8]
```diff
@@ -79,13 +80,19 @@ The following configs can be set:
...
+| `full_scan_threshold` | Minimal amount of points for additional payload-based indexing                                                                               | `None`, defaults to the default value in Qdrant* |
```
Comment (alexcg1) [inline docs/advanced/document-store/qdrant.md:93]: Minimal or minimum?

File: docs/advanced/document-store/qdrant.md [hunk 9]
```diff
@@ -79,13 +80,19 @@ The following configs can be set:
...
+| `m`                   | Number of edges per node in the index graph. Larger the value - more accurate the search, more space required                                | `None`, defaults to the default value in Qdrant* |
```
Comment (alexcg1) [inline docs/advanced/document-store/qdrant.md:94]:
```suggestion
| `m`                   | Number of edges per node in the index graph. Larger = more accurate search, more space required                                | `None`, defaults to the default value in Qdrant* |
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/qdrant.md (hunk 4 above)
Comment (AnneYang720) [reply docs/advanced/document-store/qdrant.md]: Fixed. It was an merging conflict handling error.

File: docs/advanced/document-store/qdrant.md (hunk 6 above)
Comment (AnneYang720) [reply docs/advanced/document-store/qdrant.md:87]: yeah sure :)

**Review by JoanFM — APPROVED** [review]
